clang/lib/3C/ConstraintBuilder.cpp

```diff
@@ -398,6 +399,47 @@ class FunctionVisitor : public RecursiveASTVisitor<FunctionVisitor> {
   TypeVarInfo &TVInfo;
 };
 
+class PtrToStructDef : public RecursiveASTVisitor<PtrToStructDef> {
+  public:
+    explicit PtrToStructDef(TypedefDecl *TDT) : TDT(TDT) {}
+
+    bool VisitPointerType(clang::PointerType *PT) {
+      ispointer = true;
+      return true;
+    }
+
+    bool VisitRecordType(RecordType *RT) {
+      auto decl = RT->getDecl();
+      auto declRange = decl->getSourceRange();
+      auto typedefRange = TDT->getSourceRange();
+      bool declContained = (typedefRange.getBegin() < declRange.getBegin())
+        && !(typedefRange.getEnd() < typedefRange.getEnd());
+      if (declContained) {
+        structDefInPtr = true;
```

Comment:
Rename variable: `structDefInPtr` to `structDefInTD` ? You don't know if it's in a pointer at this point, right?